wgpu-native/src/swap_chain.rs

```diff
@@ -78,6 +80,18 @@ pub struct SwapChain<B: hal::Backend> {
     pub(crate) command_pool: hal::CommandPool<B, hal::General>,
 }
 
+impl<B: hal::Backend> fmt::Debug for SwapChain<B> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SwapChain")
+            .field("raw", &self.raw)
+            .field("device_id", &self.device_id)
+            .field("desc", &self.desc)
+            .field("acquired", &self.acquired)
+            .field("sem_available", &self.sem_available)
```

Comment:
is command pool the only thing that isn't `Debug`?

Reply:
No, it's also `frames`.